src/js/player.js

```diff
@@ -1562,6 +1572,10 @@ class Player extends Component {
   techGet_(method) {
     if (this.tech_ && this.tech_.isReady_) {
 
+      if (method in middleware.allowedGetters) {
```

Comment:
This function could also use some similar cleanup, maybe we can do both in another PR, but its always bothered me that it seems so messy an nested.